### PR TITLE
fix(rpc): Fix some RPC response formats to match `zcashd`

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

As part of #4130, I discovered some differences between Zebra and `zcashd` responses.

These are some of the minor ones that were easy to fix.

### Specifications

https://zcash.github.io/rpc/getinfo.html
https://zcash.github.io/rpc/getaddressbalance.html
https://zcash.github.io/rpc/getrawmempool.html

## Solution

- Make the version number format match zcashd in getinfo
- Remove an extra array wrapper in getaddressbalance
- Return a stable sort order from getrawmempool
    - zcashd seems to return an arbitrary order here, so we might not be able to match it, and we probably shouldn't try

## Review

Anyone can review this PR, it's not urgent.

### Reviewer Checklist

  - [ ] Code implements Specs and Designs
  - [ ] Tests for Expected Behaviour
  - [ ] Tests for Errors

## Follow Up Work

- PR #4218

### CODE REVIEW

File: zebra-rpc/src/methods.rs [hunk 1]
```diff
@@ -273,8 +273,15 @@ where
     {
         let runner = Queue::start();
 
+        let mut app_version = app_version.to_string();
+
+        // Match zcashd's version format, if the version string has anything in it
+        if !app_version.is_empty() && !app_version.starts_with('v') {
+            app_version.insert(0, 'v');
```
Comment (oxarbitrage) [inline zebra-rpc/src/methods.rs:279]: I don't think we need to do this. Is lightwalletd expecting the prefix?

A version of zcash will never match a zebra one anyways.

Comment (teor2345) [reply zebra-rpc/src/methods.rs:279]: I don't particularly mind either way, but this is why I changed it:

Lightwalletd documents that `build` starts with a "v":
https://github.com/zcash/lightwalletd/blob/83bb19918fb36a5f6d5013fba7938daf3d73db1d/walletrpc/service.proto#L71

This version is sent to wallets and might be displayed to the user. So using a different format might cause wallet bugs or user confusion.

What do you think?

Comment (oxarbitrage) [reply zebra-rpc/src/methods.rs:279]: Fair enough to me.

**Review by oxarbitrage — APPROVED** [review]

File: zebra-rpc/src/methods.rs (hunk 1 above)
Comment (oxarbitrage) [reply zebra-rpc/src/methods.rs:279]: Feel free to resolve at your convenience so it can get merged.

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
Some workflows seem to have hung on this PR, perhaps due to the GitHub automatic rebase.

**Comment by teor2345** [comment]
@mergifyio refresh

**Comment by teor2345** [comment]
Temporary error:
> error: failed to solve: failed to fetch oauth token: unexpected status: 401 Unauthorized
> Error: buildx failed with: error: failed to solve: failed to fetch oauth token: unexpected status: 401 Unauthorized

**Pull request merged**